integration-cli/docker_cli_daemon_test.go

```diff
@@ -2170,6 +2170,10 @@ func (s *DockerSuite) TestDaemonDiscoveryBackendConfigReload(c *check.C) {
 
 	configFile, err = os.Create(configFilePath)
 	c.Assert(err, checker.IsNil)
+	defer func() {
+		configFile.Close()
```

Comment:
the close should go after `fmt.Fprintf(configFile, "%s", daemonConfig)` rather than at the end. The line below can be simplified with this code after moving the `Close` call:

```go
defer os.Remove(configFilePath)
```